htmd/molecule/molecule.py

```diff
@@ -882,36 +883,23 @@ def read(self, filename, type=None, skip=None, frames=None, append=False, overwr
             readers = _ALL_READERS[ext]
             for rr in readers:
                 try:
-                    to, tr = rr(fname, frame=frame, topoloc=tmppdb)
+                    mol = rr(fname, frame=frame, topoloc=tmppdb, **kwargs)
                 except FormatError:
                     continue
                 else:
                     break
 
-            if tr is not None:
-                self._keepFrame(tr, frame)
-                self._checkCoords(tr, rr, fname)
-                # TODO: Get rid of this if by moving it to a function
-                if ext in _TRAJECTORY_READERS and frame is None:
-                    # Writing hidden index file containing number of frames in trajectory file
-                    if os.path.isfile(fname):
-                        self._writeNumFrames(fname, tr.coords[0].shape[2])
-                    ff = range(np.size(tr.coords[0], 2))
-                    #tr.step = tr.step + traj[-1].step[-1] + 1
-                elif frame is None:
-                    ff = [0]
-                elif frame is not None:
-                    ff = [frame]
-                else:
-                    raise AssertionError('Should not reach here')
-                tr.fileloc = [[fname, j] for j in ff]
-                traj += tr
+            if self.numAtoms != 0 and mol.numAtoms != self.numAtoms:
```

Comment:
I wonder what the `self.numAtoms != 0` is adding to the if-clause here, but I guess it's ok

Reply:
This covers the case that the original Molecule is empty. Like doing `mol = Molecule(); mol.read(x)`. Then we want to not throw an error.